backend/inspirehep/files/api/s3.py

```diff
@@ -178,3 +205,32 @@ def create_bucket(self, bucket):
             Bucket=self.get_prefixed_bucket(bucket),
             ACL=current_app.config.get("S3_FILE_ACL"),
         )
+
+    def get_key_from_url(self, url):
+        try:
+            return urlsplit(url).path.split("/")[-1]
+        except (IndexError, AttributeError):
+            LOGGER.exception("S3 incorrect url.", url=url)
+            raise
```

Comment:
Do we want to raise twice? Isn't it the case?

Reply:
Maybe I'm wrong, but here we raise this exception only once. LOGGER.exception is logging it, but without re-raising exception wont' be propagated further.
Do you think it's better to return None here?

Reply:
👍  you're right